straxen/plugins/peak_processing.py

```diff
@@ -332,97 +334,283 @@ def find_n_competing(peaks, windows, fraction):
+                    distance = np.where(np.isnan(distance), 2 * straxen.tpc_r, distance)
+                    shadow_ref = new_shadow * 2 / (np.pi * sigma * (1 + (distance / sigma) ** 2))
+                else:
+                    shadow_ref = new_shadow
+                res['shadow_sum'][p_i] += shadow_ref
+                if near_dt < res['near_dt'][p_i]:
+                    res['near_dt'][p_i] = near_dt
+                if shadow_ref > shadow_ref_bk:
+                    shadow_ref_bk = shadow_ref
+                    res['alt_pre_x'][p_i] = res['pre_x'][p_i]
+                    res['alt_pre_y'][p_i] = res['pre_y'][p_i]
+                    res['pre_x'][p_i] = s_a['x']
+                    res['pre_y'][p_i] = s_a['y']
+                    res['alt_shadow'][p_i] = res['shadow'][p_i]
                     res['shadow'][p_i] = new_shadow
-                    res['pre_s2_area'][p_i] = s2_a['area']
-                    res['shadow_dt'][p_i] = p_a['center_time'] - s2_a['center_time']
-                    res['pre_s2_x'][p_i] = s2_a['x']
-                    res['pre_s2_y'][p_i] = s2_a['y']
+                    res['alt_pre_area'][p_i] = res['pre_area'][p_i]
+                    res['pre_area'][p_i] = s_a['area']
+                    res['alt_shadow_dt'][p_i] = res['shadow_dt'][p_i]
+                    res['shadow_dt'][p_i] = p_a['center_time'] - s_a['center_time']
+
+
+@export
+@strax.takes_config(
+    strax.Option('ambience_divide_time', default=False,
+                 help='Whether to divide area by time'),
+    strax.Option('ambience_divide_space', default=False,
+                 help='Whether to divide area by space'),
+    strax.Option(name='time_window_ambience', default=int(2e6),
+                 help='Search for ambience in this time window [ns]'),
+    strax.Option('ambience_area_parameters', default=(5, 60, 60), type=(list, tuple),
+                 help='The upper limit of S0, S1, S2 area to be counted'),
+    strax.Option(name='ambience_radius', default=6.7,
+                 help='Search for ambience in this radius [cm]'),
+)
+class PeakAmbience(strax.OverlapWindowPlugin):
+    """
+    Calculate Ambience features of peaks.
+    Features are the number of lonehits, S0, S1, S2 in a time window before peaks. 
+    References:
+        * v0.0.3 reference: xenon:xenonnt:ac:prediction:shadow_ambience
+    """
+    __version__ = '0.0.3'
+    depends_on = ('lone_hits', 'peak_basics', 'peak_positions')
+    provides = 'peak_ambience'
+    data_kind = 'peaks'
+    save_when = strax.SaveWhen.EXPLICIT
+
+    def setup(self):
+        self.ambience_divide_time = self.config['ambience_divide_time']
+        self.ambience_divide_space = self.config['ambience_divide_space']
+        self.time_window_ambience = self.config['time_window_ambience']
+        self.ambience_area_parameters = self.config['ambience_area_parameters']
+        self.ambience_radius = self.config['ambience_radius']
```

Comment:
Maybe you want to check out the new URL config by Yossi, Then you can skip these lines.

Reply:
Would you please link me to an example?

Reply:
I've found examples among codes in the master branch. I will try to use these new features.